social_engagement/handlers.py

```diff
@@ -98,20 +115,85 @@ def comment_created_signal_handler(sender, **kwargs):  # pylint: disable=unused-
 
             # update the engagement score of the thread creator
             # as well
-            task_update_user_engagement_score.delay(unicode(course_id), thread_user_id)
+            _increment(thread_user_id, course_id, 'num_comments_generated')
 
 
 @receiver(thread_followed)
+def thread_follow_signal_handler(sender, **kwargs):  # pylint: disable=unused-argument
+    """
+    Updates user social engagement score
+    """
+    thread = kwargs['post']
+    course_id = getattr(thread, 'course_id', None)
+    user_id = getattr(thread, 'user_id', None)
+    action_user = kwargs['user']  # user who followed or un-followed thread
+
+    if user_id and action_user.id != int(user_id):
+        _increment(user_id, course_id, 'num_thread_followers')
+
+
 @receiver(thread_unfollowed)
-def thread_follow_unfollow_signal_handler(sender, **kwargs):  # pylint: disable=unused-argument
+def thread_unfollow_signal_handler(sender, **kwargs):  # pylint: disable=unused-argument
     """
     Updates user social engagement score
     """
     thread = kwargs['post']
     course_id = getattr(thread, 'course_id', None)
     user_id = getattr(thread, 'user_id', None)
     action_user = kwargs['user']  # user who followed or un-followed thread
-    social_engagement_enabled = settings.FEATURES.get('ENABLE_SOCIAL_ENGAGEMENT', False)
 
-    if social_engagement_enabled and course_id and user_id and action_user.id != user_id:
-        task_update_user_engagement_score.delay(unicode(course_id), user_id)
+    if user_id and action_user.id != int(user_id):
+        # task_update_user_engagement_score.delay(unicode(course_id), user_id)
```

Comment:
I see you've deleted a lot of `.delay` calls here, but I'm wondering if that is the correct intention. Those calls would have been queued up by Celery and processed on separate machines from the LMS, which is very efficient. These new `_decrement` and `_increment` functions are running logic that happens in a single thread within the LMS, and this will thus negatively affect LMS performance. We may need to turn those calls into celery tasks, too. Let me know if you need some guidance on how to do that.